pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java

```diff
@@ -100,9 +101,13 @@ private static void validateValidationConfig(TableConfig tableConfig, @Nullable
     }
     // timeColumnName can be null in OFFLINE table
     if (timeColumnName != null && schema != null) {
-      Preconditions.checkState(schema.getSpecForTimeColumn(timeColumnName) != null,
+      FieldSpec timeColumnFieldSpec = schema.getSpecForTimeColumn(timeColumnName);
```

Comment:
There's a TableConfigUtilsTest which exhaustively tests validations done here. Can you add one for timeColumn being TIME/DATE_TIME only?